python/pyiceberg/utils/datetime.py

```diff
@@ -67,6 +67,11 @@ def time_to_micros(time_str: str) -> int:
     return (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
 
 
+def time_object_to_micros(t: time) -> int:
```

Comment:
Looks like we did a poor job naming the `time_to_micros` function because the other similar ones are `date_str_to_days` so this would be `time_str_to_micros`.

It would be nice to fix this by adding both types to `time_to_micros` and adding a `time_str_to_micros` method.